pkg/api/converterfromapi.go

```diff
@@ -221,6 +244,18 @@ func convertPropertiesToV20170131(api *Properties, p *v20170131.Properties) {
 		convertAgentPoolProfileToV20170131(apiProfile, v20170131Profile)
 		p.AgentPoolProfiles = append(p.AgentPoolProfiles, v20170131Profile)
 	}
+	if api.OrchestratorProfile.OrchestratorType == "DCOS" && len(p.AgentPoolProfiles) == 1 {
+		publicPool, _ := addDCOSPublicAgentPool(api.AgentPoolProfiles[0], api, api.MasterProfile)
+		v20170131PublicPool := &v20170131.AgentPoolProfile{}
+		// TODO verify that we want the public agent pool to be v20170131-specific
```

Comment:
I'm confused by these statements. Yes we want to have this only happen on the older versions. The newer versions allow multiple agent pools and the user can just set these themselves explicitly, so there shouldn't be a need for something like this